### PR TITLE
use bullet points for the frontpage

### PR DESCRIPTION
**Pull request by wilzbach**
rebased from #248.

While bullet points itself are already an improvement, we should make sure we put the key features concise there.

Ping @JackStouffer (you are familar with the D's feature overview, right?)

### CODE REVIEW

**Comment by stonemaster** [comment]
:+1:

File: public/content/en/welcome/welcome-to-d.md [hunk 1]
```diff
@@ -2,39 +2,31 @@
...
+- direct interface to the operating system API's and hardware
+- blazingly fast compile-times
+- allows memory-safe programming
+- _maintainable_, _easy to understand_ code.
+- evolution of C++ (without the mistakes)
+- short learning curve (similar syntax to C, C++ and Java)
+- portable code (no undefined behavior)
+- compatible with C application binary interface
+
+... and much more [features](http://dlang.org/overview.html).
```
Comment (JackStouffer) [inline public/content/en/welcome/welcome-to-d.md]: many more

**Comment by JackStouffer** [comment]
LGTM sans comments

File: public/content/en/welcome/welcome-to-d.md [hunk 2]
```diff
@@ -2,39 +2,31 @@
...
+
+- _high level language_ with great modeling power
+- _high performance_, compiled language
+- static typing
+- direct interface to the operating system API's and hardware
+- blazingly fast compile-times
+- allows memory-safe programming
+- _maintainable_, _easy to understand_ code.
+- evolution of C++ (without the mistakes)
+- short learning curve (similar syntax to C, C++ and Java)
```
Comment (JackStouffer) [inline public/content/en/welcome/welcome-to-d.md]: To not scare anyone away, I would instead say "(C-like syntax, similar to Java and JavaScript)"

File: public/content/en/welcome/welcome-to-d.md [hunk 3]
```diff
@@ -2,39 +2,31 @@
...
+- portable code (no undefined behavior)
```
Comment (JackStouffer) [inline public/content/en/welcome/welcome-to-d.md]: This isn't true. D has UB

Comment (ZombineDev) [inline public/content/en/welcome/welcome-to-d.md]: Maybe it should say "less undefined behavior"?

File: public/content/en/welcome/welcome-to-d.md [hunk 4]
```diff
@@ -2,39 +2,31 @@
...
 
 D is the culmination of _decades of experience implementing compilers_
-for many diverse languages,
-and attempting to construct large projects using those languages.
-D draws inspiration from those other languages (most especially C++)
-and tempers it with experience and real world practicality.
+for many diverse languages and has a large number of
+[unique features](http://dlang.org/overview.html):
+
+- _high level language_ with great modeling power
```
Comment (JackStouffer) [inline public/content/en/welcome/welcome-to-d.md]: D isn't a high level language, maybe instead say "high level constructs for great modeling power"

File: public/content/en/welcome/welcome-to-d.md (hunk 3 above)
Comment (wilzbach) [inline public/content/en/welcome/welcome-to-d.md]: Or should I remove the point entirely?

File: public/content/en/welcome/welcome-to-d.md (hunk 2 above)
Comment (wilzbach) [inline public/content/en/welcome/welcome-to-d.md]: Wouldn't we scare people if we mention Javascript?

File: public/content/en/welcome/welcome-to-d.md (hunk 3 above)
Comment (stonemaster) [inline public/content/en/welcome/welcome-to-d.md]: I am in favour of just dropping that point. "less undefined" may sound confusing

File: public/content/en/welcome/welcome-to-d.md (hunk 2 above)
Comment (stonemaster) [inline public/content/en/welcome/welcome-to-d.md]: I would also rather drop JavaScript. To me it appears that JS doesn't have a really good reputation among software developers. But I like @JackStouffer's suggestion. What about the compromise "(C-like syntax, similar to Java and others)"?

File: public/content/en/welcome/welcome-to-d.md (hunk 4 above)
Comment (stonemaster) [inline public/content/en/welcome/welcome-to-d.md]: I like that suggestion.

Comment (stonemaster) [inline public/content/en/welcome/welcome-to-d.md]: And: Should we maybe emphasise that D also suits well for functional programming? Something like: "D allows writing code in an imperative, object-oriented or functional style"?

**Comment by wilzbach** [comment]
Rebased with feedback incorporated (thanks!) and added these two points:

- multi-paradigm (imperative, structured, object oriented, generic, functional programming purity, and even assembly)
- built-in error prevention (contracts, unittests)

1) How should we sort those points?
2) Should we provide links for these points?

**Comment by stonemaster** [comment]
Thanks!

1.) I do like the current sorting. IMO we can just leave it the way it i.
2.) I don't think that we need links for them. We provide already a lot of links throughout the tour and we touch a lot of the topics listed. So links would just visual overhead in my eyes.

**Comment by wilzbach** [comment]
> 1.) I do like the current sorting. IMO we can just leave it the way it i.

As I need this for the D-Man on the frontpage (and it's a clear improvement), I will go ahead and merge it.
Please feel free to open a PR to make it even better!

**Comment by stonemaster** [comment]
Yep nothing is set in stone :-)

**Pull request merged**